torch_xla/csrc/tensor.cpp

```diff
@@ -120,14 +120,15 @@ XLATensor::XLATensor(std::shared_ptr<View> view,
                                        logical_element_type)) {}
 
 XLATensor::XLATensor(std::shared_ptr<Data> data)
-    : data_(std::move(data)),
+    : torch::lazy::LazyTensor(data),
+      data_(std::move(data)),
       storage_(c10::Storage(
           {}, 0,
           c10::DataPtr(nullptr, backendDeviceToAtenDevice(data_->device)))) {}
 
-XLATensor::Data* XLATensor::data() const {
-  XLA_CHECK(data_ != nullptr) << "Trying to access a null cursor";
-  return data_.get();
+auto XLATensor::data() const -> const std::shared_ptr<Data>& {
```

Comment:
nit: just curious why function declaration is used as opposed to `const std::shared_ptr<Data>& 
XLATensor::data() const`?

Reply:
The latter case you then have to specify `XLATensor::Data` instead of just using `Data`. `auto` is shorter than `XLATensor::`, and shorter is better. hahaha.